uwuno/src/test/java/com/learning/uwuno/unit/roomControllerTests.java

```diff
@@ -28,10 +31,191 @@
+        ArrayList<room> roomList = new ArrayList<>();
+        when(service.getRoomList()).thenReturn(roomList);
+        mock.perform(get("/rooms")
+                .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(content().json("[]"))
+                .andDo(print());
+    }
 
     @Test
-    public void testRoom_POST() throws Exception {
-//        this.mock.perform(post("/Rooms")).andDo(print()).andExpect(status().isOk());
+    public void GET_roomlist() throws Exception {
+        ArrayList<room> roomList = new ArrayList<>();
+        room room = new room("test", false);
+        roomList.add(room);
+        when(service.getRoomList()).thenReturn(roomList);
+
+        mock.perform(get("/rooms")
+                .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$[0]['name']").value("test"))
+                .andExpect(jsonPath("$.length()").value(1))
+                .andDo(print());
+    }
+
+    @Test
+    public void GET_room() throws Exception {
+        room room = new room("test", false);
+        when(service.getRoom("2")).thenReturn(room);
+        mock.perform(get("/rooms/2"))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.name").value(room.getName()))
+                .andExpect(jsonPath("$.uid").value(room.getUid()))
+                .andExpect(jsonPath("$.players").value(room.getPlayers()))
+                .andDo(print());
+    }
+
+    @Test
+    public void GET_room_not_found() throws Exception {
+        when(service.getRoom("2")).thenThrow(NoSuchElementException.class);
+        mock.perform(get("/rooms/2"))
+                .andExpect(status().isNotFound())
+                .andDo(print());
+    }
+
+    @Test
+    public void POST_add_room() throws Exception {
+        room room = new room("test", false);
+        when(service.addRoom(anyString(), anyBoolean())).thenReturn(room);
+
+        ArrayList<String> keys = new ArrayList<>();
+        keys.add("roomName");
+        keys.add("useBlankCards");
+
+        ArrayList<String> values = new ArrayList<>();
+        values.add("test");
+        values.add("false");
+
+        mock.perform(post("/rooms")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(testUtils.createJSON(keys, values))
+                .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+                .andExpect(jsonPath("$.name").value(room.getName()))
+                .andExpect(jsonPath("$.uid").value(room.getUid()))
+                .andDo(print());
+    }
+
+    @Test
+    public void POST_add_room_bad_req_err() throws Exception {
+        when(service.addRoom(isNull(), anyBoolean())).thenThrow(new com.learning.uwuno.errors.badRequest());
```

Comment:
I think we should be testing a bad input JSON instead of just throwing in the function and then checking if the function you mocked threw. That's testing their when().thenThrow() not our functions.
I think this applies in some of the other error tests too.

Reply:
mmm that's a good point. I missed that... I'll add some more tests zzz